### PR TITLE
[TESTING] Display rotten status when picking crafting components

### PR DESCRIPTION
**Pull request by acidia**
<a href="https://imgur.com/6xrx5k0"></a>

Mentioned by @DracoGriffin in #23937, a QOL change to let players pick the non-rotten items that are used in crafting so that you don't waste materials/time preparing rotten ingredients.  This needs a bunch of testing because items in/out of containers. items that use charges instead of qty, items held in personal inventory, inside the crafting range, and both on person and in crafting range are all handled differently and can appear in any combination. 

This won't separate rotten and unrotten items, but it will warn you if you have 3 good apples and 1 rotten one in your backpack by applying the "(rotting)" description to the group.

### CODE REVIEW

**Comment by DracoGriffin** [comment]
Travis is throwing back error on astyle but I can't find it. I got the Travis checks to pass though.

Looking forward to the change, appreciate you taking in my feature suggestion!

File: src/crafting.cpp [hunk 1]
```diff
@@ -739,29 +739,42 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
     } else { // Let the player pick which component they want to use
         uimenu cmenu;
         // Populate options with the names of the items
+        std::string rot;
```
Comment (BevapDin) [inline src/crafting.cpp]: This variable is not used outside of the loops, so there is no need to define it outside of the loops. Change the lines that reset it (`rot = "";`) to a simple definition (`std::string rot;`).

File: src/crafting.cpp [hunk 2]
```diff
@@ -739,29 +739,42 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
...
         for( auto &map_ha : map_has ) { // Index 0-(map_has.size()-1)
-            std::string tmpStr = string_format( _( "%s (%d/%d nearby)" ),
+            rot = "";
+            if( item(map_ha.type).is_comestible() && map_inv.amount_of_rotten( map_ha.type ) > 0 ){
+                rot = "(rotting)";
```
Comment (BevapDin) [inline src/crafting.cpp]: Needs translation.

**Comment by Night-Pryanik** [comment]
@acidia are you still with us? Testing is undergoing?

**Comment by kevingranade** [comment]
Seems stalled, if someone else wants to pick this up and fix up the last few issues it looks pretty close to done.

**Pull request closed without merging**